File: html/cgi/python/Cookie/welcome.py
import cgi, cgitb, os
from http.cookies import SimpleCookie

cookieString = os.environ.get('HTTP_COOKIE')
serverName = os.environ.get('SERVER_NAME')
body = "<!DOCTYPE html>"


if cookieString:
    cookie = SimpleCookie()
    cookie.load(cookieString)
    for key, morse1 in cookie.items():
        print(f"{key} is {morse1}")
    if 'bgcolor' in cookie and 'user' in cookie:
        name = cookie['user'].value
        bgcolor = cookie['bgcolor'].value
        body += "<head><title>"
        body += name
        body += "</title><style>* {font-family: system-ui, -apple-system, BlinkMacSystemFont,"
        body += "'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;}"
        body += "body { background-color: "
        body += bgcolor
        body += "; display: flex; justify-content: center; align-items: center; height: 100vh;"
        body += "margin: 0; flex-direction: column;} h1 { font-size: 3em; color: white;"
        body += "} </style> </head>"
        body += "<body><h1>"
        body += name
        body += "</h1><h2>Welcome to your panel !</h2></body></html>"
        body += "<br><a href=\"/python/ColorModifier.html\"><button> Change Your Profile Color</button></a>"
        body += "<br><a href=\"/python/LogOut.py\"><button> Log Out </button></a>"
else:
    body += "<head><title> Unknown User"
    body += "</title><style>* {font-family: system-ui, -apple-system, BlinkMacSystemFont,"
    body += "'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;}"
    body += "body { background-color: "
    body += "LightGoldenRodYellow"
    body += "; display: flex; justify-content: center; align-items: center; height: 100vh;"
    body += "margin: 0; flex-direction: column;} h1 { font-size: 3em; color: black;"
    body += "} </style> </head>"
    body += "<body><h1>"
    body += "Oops ! You're not connected yet"
    body += "</h1></body></html>"
    body += "<br><a href=\"/python/connectionForm.html\"><button> Create a new profile</button></a>"

print("Content-Type: Text/html")
print(f"Server: {serverName}")
print(f"Content-Length: {len(body)}\r\n")
print(f"{body}")

File: html/cgi/php/Cookie/welcome.php
<?php
// Enable error reporting for debugging
error_reporting(E_ALL);
ini_set('display_errors', 1);

// Function to print cookies (equivalent to the print statements in the Python script)
function print_cookies($cookies) {
    foreach ($cookies as $key => $value) {
        echo "$key is $value<br>";
    }
}

// Retrieve cookies and server name
$cookies = $_COOKIE;
$serverName = $_SERVER['SERVER_NAME'] ?? '';

// Initialize the HTML body
$body = "<!DOCTYPE html>";

if (!empty($cookies)) {
    print_cookies($cookies);

    if (isset($cookies['bgcolor']) && isset($cookies['user'])) {
        $name = htmlspecialchars($cookies['user']);
        $bgcolor = htmlspecialchars($cookies['bgcolor']);
        $body .= "<head><title>";
        $body .= $name;
        $body .= "</title><style>* {font-family: system-ui, -apple-system, BlinkMacSystemFont,"
            . "'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;}"
            . "body { background-color: "
            . $bgcolor
            . "; display: flex; justify-content: center; align-items: center; height: 100vh;"
            . "margin: 0; flex-direction: column;} h1 { font-size: 3em; color: white;}"
            . "</style></head><body><h1>"
            . $name
            . "</h1><h2>Welcome to your panel!</h2><br><a href=\"/php/ColorModifier.html\"><button> Change Your Profile Color</button></a>"
            . "<br><a href=\"/php/LogOut.php\"><button> Log Out </button></a></body></html>";
    } else {
        $body .= "<head><title> Unknown User</title><style>* {font-family: system-ui, -apple-system, BlinkMacSystemFont,"
            . "'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;}"
            . "body { background-color: LightGoldenRodYellow; display: flex; justify-content: center; align-items: center; height: 100vh;"
            . "margin: 0; flex-direction: column;} h1 { font-size: 3em; color: black;}"
            . "</style></head><body><h1>Oops! You're not connected yet</h1>"
            . "<br><a href=\"/php/connectionForm.html\"><button> Create a new profile</button></a></body></html>";
    }
} else {
    $body .= "<head><title> Unknown User</title><style>* {font-family: system-ui, -apple-system, BlinkMacSystemFont,"
        . "'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;}"
        . "body { background-color: LightGoldenRodYellow; display: flex; justify-content: center; align-items: center; height: 100vh;"
        . "margin: 0; flex-direction: column;} h1 { font-size: 3em; color: black;}"
        . "</style></head><body><h1>Oops! You're not connected yet</h1>"
        . "<br><a href=\"/php/connectionForm.html\"><button> Create a new profile</button></a></body></html>";
}

// Print the HTTP headers
header("Content-Type: text/html");
header("Content-Length: " . strlen($body));

// Print the server name and the HTML body
echo "Server: $serverName<br>";
echo $body;
?>
